server-common/src/main/java/org/apache/kafka/purgatory/Purgatory.java

```diff
@@ -26,10 +26,9 @@
 
 /**
  * The purgatory which holds events that have been started, but not yet completed.
- * We wait for the high water mark of the metadata log to advance before completing
- * them.
+ * We wait for the high watermark of the log to advance before completing them.
  */
-class ControllerPurgatory {
+public class Purgatory {
```

Comment:
It seems better to add a prefix since we have several `Purgatory` classes, we have a `FuturePurgatory` in raft module and its similar to this `ControllerPurgatory`, we may call it `DeferredEventPurgatory` or HwmPurgatory(high watermark).

Reply:
+1 for `DeferredEventPurgatory`.